### PR TITLE
Add Dynatrace RUM

### PR DESCRIPTION
**Pull request by kyle-ssg**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/deployment/locally-api#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?

## Changes

_Please describe._

## How did you test this code?

_Please describe._

### CODE REVIEW

File: frontend/api/index.js [hunk 1]
```diff
@@ -111,14 +111,86 @@ app.get('/config/project-overrides', (req, res) => {
       value: envToBool('DISABLE_INVITE_LINKS', false),
     },
   ]
-  const output = values.map(getVariable).join('')
-
+  let output = values.map(getVariable).join('')
+  let dynatrace = ''
+  if (process.env.DYNATRACE_URL) {
```
Comment (kyle-ssg) [inline frontend/api/index.js:116]: If DYNATRACE_URL env var is set, add the following snippet to the end of project-overrides:

Comment (kyle-ssg) [reply frontend/api/index.js:116]: Note in the snippet, reportUrl=${process.env.DYNATRACE_URL} - it will need to do this too

Comment (matthewelwell) [reply frontend/api/index.js:116]: This is uber skank but not sure if there is an alternative?

File: frontend/api/index.js [hunk 2]
```diff
@@ -111,14 +111,86 @@ app.get('/config/project-overrides', (req, res) => {
...
+                    !Lg){var Ef=Ga("agentUri")||ad(Ga("featureHash"));if(W("async")||"complete"===document.readyState)Zc(Ef,W("async"),void 0,void 0,"dtjsagent");else{var Oi="".concat("dtjsagent","dw");document.write('<script id="'.concat(Oi,'" type="text/javascript" src="').concat(Ef,'">\x3c/script>'));document.getElementById(Oi)||Zc(Ef,W("async"),void 0,void 0,"dtjsagent")}}Za("dtCookie")&&Cb("cooO",!0);zf();Cb("pVO",!!nb("dt-pVO"));Wb(fc);cg=18E5;Gg=oa("hvt")||216E5;jb(Fd,[1])}catch(Mh){delete Wa.dT_,fb()&&Wa.console.log("JsAgent initCode initialization failed!",
+                Mh)}}})()})();`
+  }
   res.setHeader('Cache-Control', 's-max-age=1, stale-while-revalidate')
   res.setHeader('content-type', 'application/javascript')
   res.send(`window.projectOverrides = {
         ${output}
     };
+    
+    ${dynatrace}
```
Comment (matthewelwell) [inline frontend/api/index.js:190]: Should this be included in the `res.send()` ?

File: frontend/web/project/api.js [hunk 3]
```diff
@@ -44,6 +44,10 @@ global.API = {
       mixpanel.alias(id)
     }
 
+    if (typeof DYNATRACE_URL !== 'undefined' && !!DYNATRACE_URL) {
```
Comment (matthewelwell) [inline frontend/web/project/api.js]: I'd prefer to see this code not written out multiple times, can we use the `enableDynatrace` var from a central location somehow? Or just redefine it here so we're at least only doing the check once.

**Review by matthewelwell — APPROVED** [review]
I've approved it, but added a few minor comments.

File: frontend/api/index.js (hunk 1 above)
Comment (kyle-ssg) [reply frontend/api/index.js:116]: There isnt

File: frontend/api/index.js (hunk 2 above)
Comment (kyle-ssg) [reply frontend/api/index.js:190]: Yeah we want it to return the dynatrace script

File: frontend/web/project/api.js (hunk 3 above)
Comment (kyle-ssg) [reply frontend/web/project/api.js]: Done

File: frontend/api/index.js (hunk 2 above)
Comment (matthewelwell) [reply frontend/api/index.js:190]: Sorry, got confused by the formatting, my bad.

**Pull request merged**